src/organisms/Address/Address.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ChangeEvent, Component, FormEvent } from 'react';
+
+import styled from '_styled-components';
+import { Button, Identicon } from 'atoms';
+import { Copyable } from 'molecules';
+import { borderRadius, scale } from 'Theme';
+import Typography from 'Typography';
+
+const Flex = styled.div`
+  display: flex;
+`;
+
+const Content = styled.div`
+  padding-left: ${scale(0)};
+`;
+
+const Title = styled(Typography)<{ clickable: boolean }>`
+  display: inline;
+  font-size: ${scale(1)};
+  ${props => props.clickable && `cursor: pointer;`};
+`;
+
+Title.defaultProps = { as: 'div' };
+
+const TitleInput = styled.input`
+  background: ${props => props.theme.background};
+  border: 0.125em solid ${props => props.theme.controlBorder};
+  border-radius: ${borderRadius};
+  color: ${props => props.theme.text};
+  font: ${scale(1)} Lato, sans-serif;
+  padding: 0;
+`;
+
+const ColoredIconButton = styled(Button)`
+  color: #b5bfc7;
+`;
+
+const SubmitButton = styled(ColoredIconButton)`
+  color: #1eb8e7;
+  left: -${scale(2)};
+  position: relative;
+`;
+
+SubmitButton.defaultProps = { type: 'submit', icon: 'check' };
+
+interface Props {
+  address: string;
+  title: string;
+  onSubmit?(title: string): void;
+  truncate?(text: string): string;
+}
+
+interface State {
+  editing: boolean;
+  title: string;
+}
+
+export class Address extends Component<Props, State> {
+  public constructor(props: Props) {
+    super(props);
+    this.state = { editing: false, title: props.title };
+  }
+
+  public handleChange = ({
+    target: { value },
+  }: ChangeEvent<HTMLInputElement>) => this.setState({ title: value });
+
+  public handleEditing = () => this.setState({ editing: true });
+
+  public handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const { onSubmit } = this.props;
+    const { title } = this.state;
+
+    this.setState({ editing: false });
+    onSubmit!(title);
```

Comment:
What does the exclamation mark do here?

Reply:
`!` casts something to non-null/undefined. If I didn't do this, TypeScript would error because onSubmit is a function or undefined, and undefined isn't callable. Note that this is a feature of TypeScript that's being disabled, so it's technically less safe, though I know the `onSubmit` prop probably exists if `editing` was `true`.

Reply:
Ahh ok I'll have to keep this in mind for similar situations. Could this be used in places where we did focus checking?

Reply:
I'm not sure where you mean, do you have an example?